### PR TITLE
[Graph] Refactored creation of nodes

### LINKED ISSUE #549: Graph format: $node variable is not initialized in the else block
### Setup

- MW version: 1.31.5
- SMW version: 3.1.0
- SRF version: 3.2.0-alpha

### Issue

Error from line 186 of /var/www/smw/extensions/SemanticResultFormats/src/Graph/GraphPrinter.php: Call to a member function addParentNode() on null

### PR DESCRIPTION
**Pull request by gesinn-it-gea**
Refactored creation of nodes. This also fixes #549.

### CODE REVIEW

**Comment by gesinn-it** [comment]
@JeroenDeDauw can you please have a look at this PR. Travis still reports one failed test for Gantt as described in #575.

**Comment by gesinn-it** [comment]
@kghbln @JeroenDeDauw can this PR be merged?

**Comment by gesinn-it** [comment]
Anything we can do to help this getting merged? Anything missing?

File: src/Graph/GraphPrinter.php [hunk 1]
```diff
@@ -174,13 +174,28 @@ protected function processResultRow( array /* of SMWResultArray */ $row ) {
...
+				// check, if node has already been created
+				/** @var GraphNode $node */
+				foreach ( $this->nodes as $node ) {
+					if ( $node->getID() == $id ) {
+						$nodeExists = true;
+						break;
+					}
+				}
+
+				if ( !$nodeExists ) {
```
Comment (JeroenDeDauw) [inline src/Graph/GraphPrinter.php:189]:
```
if ( !$this->nodeExists( $id ) ) {
```

Which itself could go in a 

```
$this->newGrapthNode( $object->getShortWikiText() )
```

Comment (gesinn-it) [reply src/Graph/GraphPrinter.php:189]: Sorry, @JeroenDeDauw what do you mean with your comment?

Comment (krabina) [reply src/Graph/GraphPrinter.php:189]: any progress here? @JeroenDeDauw @gesinn-it-gea ?

**Comment by gesinn-it-gea** [comment]
We will check against current dev-master today. If everything works as expected, I'll merge it.

**Comment by alex-mashin** [comment]
I think, this PR can be closed, after  #749.

**Comment by JeroenDeDauw** [comment]
@gesinn-it-gea just say so if you think closing is incorrect

**Pull request closed without merging**